src/ILLink.RoslynAnalyzer/RequiresUnreferencedCodeAnalyzer.cs

```diff
@@ -76,26 +76,22 @@ public override void Initialize (AnalysisContext context)
 					IMethodSymbol method,
 					Location location)
 				{
-					var attributes = method.GetAttributes ();
-
-					foreach (var attr in attributes) {
-						if (attr.AttributeClass is { } attrClass &&
-							IsNamedType (attrClass, "System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute") &&
-							attr.ConstructorArguments.Length == 1 &&
-							attr.ConstructorArguments[0] is { Type: { SpecialType: SpecialType.System_String } } ctorArg) {
-
-							string urlArgument = "";
-							foreach (var namedArgument in attr.NamedArguments) {
-								if (namedArgument.Key == "Url")
-									urlArgument = namedArgument.Value.Value?.ToString () ?? "";
-							}
-							operationContext.ReportDiagnostic (Diagnostic.Create (
-								s_rule,
-								location,
-								method.OriginalDefinition.ToString (),
-								(string) ctorArg.Value!,
-								urlArgument));
+					AttributeData? attribute = GetRequiresUnreferencedCodeAttribute (operationContext.ContainingSymbol.GetAttributes ());
+					if (attribute != null)
+						return;
+					attribute = GetRequiresUnreferencedCodeAttribute (method.GetAttributes ());
+					if (attribute != null) {
+						string urlArgument = "";
+						foreach (var namedArgument in attribute.NamedArguments) {
+							if (namedArgument.Key == "Url")
+								urlArgument = namedArgument.Value.Value?.ToString () ?? "";
```

Comment:
nit: why not use this in the declaration of `urlArgument`? Something like `attribute.NamedArguments.FirstOrDefault (na => na.Key == "Url").Value.Value?.ToString ()` then we don't need the `foreach`.